src/server.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+  const parts = normalizedURL.pathname.slice(1).split('/');
+  const query = {};
+
+  for (const [key, value] of normalizedURL.searchParams.entries()) {
+    query[key] = value;
+  }
+
+  if (!parts.length && !query.length) {
```

Comment:
`query` is an object. You're reading its field `length` which in your example is undefined. You can check it by `Object.keys(query).length`